app/src/main/java/com/example/concordia_campus_guide/Activities/PathsViewModel.java

```diff
@@ -16,40 +16,40 @@ public Place getFrom() {
         return SelectingToFromState.getFrom();
     }
 
-    public Place getEntrance(Place place){
+    public Place getEntrance(Place place) {
         if (place instanceof RoomModel) {
             String campusCode = ((RoomModel) place).getCampus();
             String floorCode = ((RoomModel) place).getFloorCode();
-            String buildingCode = floorCode.substring(0, floorCode.indexOf("-")).toUpperCase();
+            String buildingCode = floorCode.substring(0, floorCode.indexOf('-')).toUpperCase();
             //For indoor direction, no need to return the actual room from the db as the room coordinates will not be used by the PathFinder
             return new RoomModel(place.getCenterCoordinates(), "entrance", buildingCode + "-1", campusCode);
         }
         return place;
     }
 
-    public boolean arePlacesSeparatedByATunnel(Place from, Place to){
-        if(from instanceof RoomModel && to instanceof RoomModel){
+    public boolean arePlacesSeparatedByATunnel(Place from, Place to) {
+        if (from instanceof RoomModel && to instanceof RoomModel) {
             String floorCode = ((RoomModel) from).getFloorCode();
-            String from_building = floorCode.toUpperCase().substring(0, floorCode.indexOf("-"));
+            String fromBuilding = floorCode.toUpperCase().substring(0, floorCode.indexOf('-'));
 
             floorCode = ((RoomModel) to).getFloorCode();
-            String to_building = floorCode.toUpperCase().substring(0, floorCode.indexOf("-"));
+            String toBuilding = floorCode.toUpperCase().substring(0, floorCode.indexOf('-'));
 
-            return from_building.equalsIgnoreCase("H") && to_building.equalsIgnoreCase("MB") ||
-                    from_building.equalsIgnoreCase("MB") && to_building.equalsIgnoreCase("H");
+            return fromBuilding.equalsIgnoreCase("H") && toBuilding.equalsIgnoreCase("MB") ||
+                    fromBuilding.equalsIgnoreCase("MB") && toBuilding.equalsIgnoreCase("H");
         }
         return false;
     }
```

Comment:
I am not sure if i understand the method correctly. I am just curious, what if the places to and from are actually buildings or floors. the if statment won't be executed and the method will return false even if the two floors are from two different buildings. Is it the case?

Reply:
Ummm interesting... Viveka wrote this method so I think she can correct me. Yeah what you said is correct so if you try to go from Hall building to mb-s2 it will give you outdoor directions rather than the tunnel. It's not wrong execution I'm just not sure if that's what we want